ingredient/ingredient-function-app/src/arm.json

```diff
@@ -0,0 +1,72 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+      "appName": {
+        "type": "string",
+        "metadata": {
+          "description": "The name of the function app that you wish to create."
+        }
+      },
+      "planName": {
+        "type": "string",
+        "metadata": {
+          "description": "The name of the hosting plan you wish to associate the function app with."
+        }
+      },
+      "storageAccountName": {
+        "type": "string",
+        "metadata": {
+          "description": "The name of the storage account you wish to associate the function app with."
+        }
+      },
+      "location": {
+        "type": "string",
+        "defaultValue": "[resourceGroup().location]",
+        "metadata": {
+          "description": "Location for all resources."
+        }
+      },
+      "appInsightsName": {
+        "type": "string",
+        "metadata": {
+          "description": "Application Insights resource for monitoring."
+        }
+      }
+    },
+    "variables": {},
+    "resources": [
+      {
+        "apiVersion": "2016-08-01",
+        "type": "Microsoft.Web/sites",
+        "name": "[parameters('appName')]",
+        "location": "[parameters('location')]",
+        "kind": "functionapp",
+        "properties": {
+          "name": "[parameters('appName')]",
+          "serverFarmId": "[resourceId('Microsoft.Web/serverfarms', parameters('planName'))]",
+          "hostingEnvironment": "",
+          "clientAffinityEnabled": false,
+          "siteConfig": {
+            "alwaysOn": true
+          }
+        },
+        "resources": [
+          {
+            "apiVersion": "2016-08-01",
+            "name": "appsettings",
+            "type": "config",
+            "dependsOn": [
+              "[resourceId('Microsoft.Web/sites', parameters('appName'))]"
+            ],
+            "properties": {
+              "AzureWebJobsStorage": "[concat('DefaultEndpointsProtocol=https;AccountName=',parameters('storageAccountName'),';AccountKey=',listkeys(resourceId('Microsoft.Storage/storageAccounts', parameters('storageAccountName')), '2015-05-01-preview').key1,';')]",
```

Comment:
Have you tried this using resources outside the resource group where the function app is deploying to?

More then likely the storage/host/AI resources will be shared and in another resource group. We might need to pass in both account name & rg for each of these. The resourceId() function lets you pass in a RG name as an optional param.

Reply:
If you do have to pass in an RG & reousrce name for storage/ai/etc.. The input params should follow a pattern of "<rg>/<resource>" or "<resource>". In your plugin parse the param for rg & resource. if rg isn't supplied, assume rg of the package which you can lookup with coreutils. @JasonPagel does this for some of his ingredients. Not sure if that "group/resource" parser made it into core, or part of the platform.

Reply:
sounds like I would need 3 more ARM params for the rg's (unless you have a more elegant idea?)

Reply:
yeah 3 more ARM params. Think you an keep it 3 params in bake though and split it out in code.

Reply:
If you want to push forward with the current limitation, create an open issue and we can track it as a next version.